src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -455,34 +431,34 @@ else if (conf.disk_access_mode == Config.DiskAccessMode.mmap_index_only)
         if (conf.concurrent_replicates != null)
             logger.warn("concurrent_replicates has been deprecated and should be removed from cassandra.yaml");
 
-        if (conf.networking_cache_size_in_mb == null)
-            conf.networking_cache_size_in_mb = Math.min(128, (int) (Runtime.getRuntime().maxMemory() / (16 * 1048576)));
+        if (conf.networking_cache_size == null)
+            conf.networking_cache_size = DataStorage.inMegabytes(Math.min(128, (Runtime.getRuntime().maxMemory() / (16 * 1048576))));
 
-        if (conf.file_cache_size_in_mb == null)
-            conf.file_cache_size_in_mb = Math.min(512, (int) (Runtime.getRuntime().maxMemory() / (4 * 1048576)));
+        if (conf.file_cache_size == null)
+            conf.file_cache_size = DataStorage.inMegabytes(Math.min(512, (Runtime.getRuntime().maxMemory() / (4 * 1048576))));
 
         // round down for SSDs and round up for spinning disks
         if (conf.file_cache_round_up == null)
             conf.file_cache_round_up = conf.disk_optimization_strategy == Config.DiskOptimizationStrategy.spinning;
 
-        if (conf.memtable_offheap_space_in_mb == null)
-            conf.memtable_offheap_space_in_mb = (int) (Runtime.getRuntime().maxMemory() / (4 * 1048576));
-        if (conf.memtable_offheap_space_in_mb < 0)
-            throw new ConfigurationException("memtable_offheap_space_in_mb must be positive, but was " + conf.memtable_offheap_space_in_mb, false);
+        if (conf.memtable_offheap_space == null)
+            conf.memtable_offheap_space = DataStorage.inMegabytes(Runtime.getRuntime().maxMemory() / (4 * 1048576));
+        if (conf.memtable_offheap_space.toMegabytes() < 0)
```

Comment:
This can be negative because `DataStorage.inMegabytes()` preserves the negative value? (Should it?)

Reply:
With the change we mentioned last night(block on negative in general for` DataStorage` and `BitRate` the way we decided for `CassandraDuration`), this should be gone. No negative `DataStorage`